v2/dataplex/src/main/java/com/google/cloud/teleport/v2/options/DataplexBigQueryToGcsOptions.java

```diff
@@ -75,12 +75,9 @@ public interface DataplexBigQueryToGcsOptions
 
   void setDestinationStorageBucketAssetName(String destinationStorageBucketAssetName);
 
-  @TemplateParameter.Text(
+  @TemplateParameter.DateTime(
```

Comment:
I'm not sure we should use the DateTime type.
I don't know why, but for some reason it was added as a separate type in the templates plugin. The problem is that the API doesn't actually support DateTime as the parameter type, so the plugin translates it to the regular TEXT parameter: https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/ffc6262b6fb66a30069b18bb1fe5c34ade117535/plugins/core-plugin/src/main/java/com/google/cloud/teleport/plugin/model/ImageSpecParameter.java#L365

So by switching from Text to DateTime here we achieved nothing and only lost the regexp check (so now a user can enter anything into this text field without the format being checked).

I think we should either:
1) Keep using Text type here and in other similar places, or
2) Switch to DateTime, but add the `regexes` attribute to this annotation too, and translate it to TEXT param with a regexp in the plugin. I think different templates can expect dates to be in different format, so we can't just use a common hard-coded pattern.

Reply:
Could you please take a look [here](https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/ffc6262b6fb66a30069b18bb1fe5c34ade117535/metadata/src/main/java/com/google/cloud/teleport/metadata/util/MetadataUtils.java#L188), do we really lose regex with `DateTime`?

Reply:
I agree that it is safer not to change `DateTime` here because it had its own regex. But in other places there was only `Text` without regex, so by changing to `DateTime` we will at least add some common regex. What do you think?

Reply:
> Could you please take a look [here](https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/ffc6262b6fb66a30069b18bb1fe5c34ade117535/metadata/src/main/java/com/google/cloud/teleport/metadata/util/MetadataUtils.java#L188), do we really lose regex with DateTime?

Yes, good point, I missed that. Turns out there is a default regexp for DateTime.
But still, switching from Text with a regexp that is different from this default to DateTime can be problematic.

> But in other places there was only Text without regex, so by changing to DateTime we will at least add some common regex. What do you think?

It's possible that for some of these parameters there is no regexp in the annotation (and hence the format is not checked in the UI), but a certain format is checked/expected in the pipeline itself (it'll just fail in runtime if it's wrong). So my enforcing the DateTime default regexp here we can run into problems with this format not being compatible with what a template expects.
I think we can change it to DateTime only if we review and test each of these templates to make sure the default DateTime format is actually what it expects.
I'd err on the side of just not changing this for now, it's not very important.